component/src/main/java/org/wso2/extension/siddhi/io/feed/sink/FeedSink.java

```diff
@@ -0,0 +1,268 @@
+                        description = "time out of the atom http response, in milliseconds",
+                        optional = true,
+                        defaultValue = Constants.DEFAULT_TIME_OUT,
+                        type = DataType.INT)
+        },
+        examples = {
+                @Example(
+                        syntax = "@App:name('test')n" +
+                                "@sink(type='feed',\n" +
+                                "url = 'localhost:8080/news',\n" +
+                                "http.response.code = '202',\n" +
+                                "@map(type = 'keyvalue', fail.on.missing.attribute = 'false'))\n" +
+                                " define stream outputStream(content string, title string);",
+                        description = " This example shows how to create Atom entry on existing atom document. The " +
+                                "variables of stream are the standard element of a atom entry "
+                )
+        }
+)
+
+public class FeedSink extends Sink {
+
+    private URL url;
+    private Abdera abdera;
+    private AbderaClient abderaClient;
+    private int httpResponse;
+    private String atomFunc;
+    private StreamDefinition streamDefinition;
+    private OptionHolder optionHolder;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+            return new Class[]{Map.class};
+    }
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+            return new String[0];
+    }
+
+    @Override
+    protected void init(StreamDefinition streamDefinition, OptionHolder optionHolder, ConfigReader configReader,
+            SiddhiAppContext siddhiAppContext) {
+        try {
+            this.url = new URL(optionHolder.validateAndGetStaticValue(Constants.URL));
+        } catch (MalformedURLException e) {
+            throw new SiddhiAppValidationException("Url Syntax Error in " + streamDefinition.getId() + ". Given value "
+                    + optionHolder.validateAndGetStaticValue(Constants.URL));
+        }
+        this.optionHolder = optionHolder;
+        this.streamDefinition = streamDefinition;
+        this.abdera = new Abdera();
+        this.abderaClient = new AbderaClient(abdera);
+        this.httpResponse = Integer.parseInt(optionHolder.validateAndGetStaticValue(Constants.HTTP_RESPONSE_CODE,
+                Constants.HTTP_CREATED));
+        this.atomFunc = validateAtom(optionHolder.validateAndGetStaticValue(Constants.ATOM_FUNC,
+                Constants.FEED_CREATE));
+
+        int timeout = validateTimeOut();
+        this.abderaClient.setConnectionTimeout(timeout);
+        String userName = optionHolder.validateAndGetStaticValue(Constants.USERNAME, Constants.CREDENTIALS);
+        String password = optionHolder.validateAndGetStaticValue(Constants.PASSWORD, Constants.CREDENTIALS);
+        if (!optionHolder.validateAndGetStaticValue(Constants.USERNAME, Constants.CREDENTIALS)
```

Comment:
No need to optionHolder.validateAndGetStaticValue() within the condition. Use the value you've already extracted to check against the user given credentials to be the default ones.
If the username and password are not case-sensitive, instead of using .equals() use equalsIgnoreCase()